modules/prompt/functions/async

```diff
@@ -1 +1 @@
-../external/pure/async.zsh
\ No newline at end of file
+../external/async/async.zsh
```

Comment:
You said that your issue was fixed when you restarted the shell? My guess is that zsh cached the symlink resolution so it had issues when the symlink was changed and the submodule wasn't updated.

The preferred `async` (at least in my opinion) is the external dependency, not the one bundled with pure. The one bundled with pure is meant as a fallback.